File: prakash/19_AnagramChecker/AnagramChecker.test.js
// anagramChecker.test.ts
const areAnagrams = require("./AnagramChecker");
describe("Anagram Checker", () => {
  test("Test with valid anagrams", () => {
    expect(areAnagrams("listen", "silent")).toBe(true);
    expect(areAnagrams("evil", "vile")).toBe(true);
    expect(areAnagrams("elbow", "below")).toBe(true);
  });

  test("Test with strings that are not anagrams", () => {
    expect(areAnagrams("hello", "world")).toBe(false);
    expect(areAnagrams("cat", "dog")).toBe(false);
    expect(areAnagrams("abc", "def")).toBe(false);
  });

  test("Test with strings of different lengths", () => {
    expect(areAnagrams("abc", "abcd")).toBe(false);
    expect(areAnagrams("a", "aa")).toBe(false);
  });

  test("Test with empty strings", () => {
    expect(areAnagrams("", "")).toBe(true);
  });
});

File: Bipin/019_anagram_checker/anagram_checker_test.go
package main

import "testing"

// TestAreAnagrams contains unit tests for the AnagramChecker function.
func TestAnagramChecker(t *testing.T) {
	// 1. Test with valid anagrams
	if !AnagramChecker("Debit Card", "Bad Credit") {
		t.Errorf("Expected true, got false for 'Debit Card' and 'Bad Credit'")
	}

	// 2. Test with strings that are not anagrams
	if AnagramChecker("hello", "world") {
		t.Errorf("Expected false, got true for 'hello' and 'world'")
	}

	// 3. Test with strings of different lengths
	if AnagramChecker("abc", "abcd") {
		t.Errorf("Expected false, got true for 'abc' and 'abcd'")
	}

	// 4. Test with empty strings
	if !AnagramChecker("", "") {
		t.Errorf("Expected true, got false for two empty strings")
	}
}
